amazonclouddrive/amazonclouddrive.go

```diff
@@ -668,6 +668,51 @@ func (f *Fs) Hashes() fs.HashSet {
 	return fs.HashSet(fs.HashMD5)
 }
 
+func (f *Fs) Move(src fs.Object, remote string) (fs.Object, error) {
+	srcObj, ok := src.(*Object)
+	if !ok {
+		fs.Debug(src, "Can't move - not same remote type")
+		return nil, fs.ErrorCantMove
+	}
+
+	// Temporary Object under construction
+	dstObj := &Object{
+		fs:     f,
+		remote: remote,
+	}
+
+	var err error
+	_, directoryID, err := f.dirCache.FindPath(remote, false)
+	if err != nil {
+		return nil, err
+	}
+
+	var info *acd.Node
+	var resp *http.Response
+	if directoryID == srcObj.info.Parents[0] {
+		// Do the rename
+		err = f.pacer.Call(func() (bool, error) {
+			info, resp, err = srcObj.info.Rename(remote)
+			return srcObj.fs.shouldRetry(resp, err)
+		})
+		if err != nil {
+			return nil, err
+		}
+	} else {
+		// Do the move
+		err = f.pacer.Call(func() (bool, error) {
+			info, resp, err = srcObj.info.Move(directoryID)
```

Comment:
This is ignoring the fact that `remote` as passed in to `Move` might have a different leaf to `srcObj.remote` isn't it?

`f.dirCache.FindPath` returns the new leaf too so I suspect you want to capture that in a variable and use it here.  I don't know whether you can rename the object as well as change its parents in one call though...